src/OpenTelemetry.Api/Trace/Status.cs

```diff
@@ -168,13 +168,21 @@ internal Status(CanonicalCode canonicalCode, string description = null)
         /// <summary>
         /// Gets a value indicating whether span completed sucessfully.
         /// </summary>
-        public bool IsOk
-        {
-            get
-            {
-                return this.CanonicalCode == CanonicalCode.Ok;
-            }
-        }
+        public bool IsOk => this.CanonicalCode == CanonicalCode.Ok;
+
+        /// <summary>
+        /// Compare two <see cref="Status"/> for equality.
+        /// </summary>
+        /// <param name="status1">First Status to compare.</param>
+        /// <param name="status2">Second Status to compare.</param>
+        public static bool operator ==(Status status1, Status status2) => status1.Equals(status2);
```

Comment:
is it just for a best practice or has some practical use?

Reply:
both, you can't == structs until you override it 

http://csharphelper.com/blog/2017/08/use-and-to-compare-structs-in-c/